src/go/vtunnel/test/e2e/connectivity_test.go

```diff
@@ -0,0 +1,220 @@
+
+	defer func() {
+		t.Logf("Deleting %v wsl distro", wslDistroName)
+		unregisterCmd := cmdExec("", "wsl", "--unregister", wslDistroName)
+		err := unregisterCmd.Run()
+		require.NoErrorf(t, err, "Failed to unregister distro %v", wslDistroName)
+	}()
+
+	// It takes a long time to start a new distro,
+	// 20 sec is a long time but that's actually how long
+	// it takes to start a distro without any flakiness
+	timeout := time.Second * 20
+	tryInterval := time.Second * 2
+	err = confirm(func() bool {
+		// Run `wslpath` to see if the distribution is registered; this avoids
+		// parsing the output of `wsl --list` to avoid having to handle UTF-16.
+		out, err := cmdRunWithOutput("wsl", "--distribution", wslDistroName, "--exec", "/bin/wslpath", ".")
+		if err != nil {
+			return false
+		}
+		// We expect `wslpath` to output a single dot for the given command.
+		return strings.TrimSpace(out) == "."
+	}, tryInterval, timeout)
+	require.NoErrorf(t, err, "Failed to check if %v wsl distro is running", wslDistroName)
+
+	t.Logf("Starting a test HTTP server in the host machine listening on [%v]", testHTTPServerAddr)
+	go startHTTPServer(t, testHTTPServerAddr)
+
+	t.Logf("Starting vtunnel peer process in wsl [%v]", wslDistroName)
+	peerCmd := cmdExec(
+		tmpDir,
+		"wsl", "--user", "root",
+		"--distribution", wslDistroName,
+		"--exec", "./main", "peer",
+		"--handshake-port", handShakePort,
+		"--vsock-port", vSockHostPort,
+		"--listen-address", fmt.Sprintf("127.0.0.1:%s", peerTCPPort))
+	err = peerCmd.Start()
+	require.NoError(t, err, "Starting vtunnel peer process faild")
+	defer func() {
+		_ = peerCmd.Process.Kill()
+	}()
+
+	t.Log("Starting vtunnel host process")
+	vtunHostPath := filepath.Join(tmpDir, "main.exe")
+	hostCmd := cmdExec(
+		tmpDir,
+		vtunHostPath, "host",
+		"--upstream-address", testHTTPServerAddr,
+		"--handshake-port", handShakePort,
+		"--vsock-port", vSockHostPort)
+	err = hostCmd.Start()
+	require.NoError(t, err, "Starting vtunnel host process faild")
+	defer func() {
+		_ = hostCmd.Process.Kill()
+	}()
+
+	t.Log("Confirming vtunnel peer process is up")
+	peerCmdTimeout := time.Second * 10
+	err = confirm(func() bool {
+		p, err := os.FindProcess(peerCmd.Process.Pid)
+		if err != nil {
+			t.Logf("looking for vtunnel peer process PID: %v", err)
+			return false
+		}
+		return p.Pid == peerCmd.Process.Pid
+	}, tryInterval, peerCmdTimeout)
+	require.NoError(t, err, "failed to confirm vtunnel peer process is running")
+
+	t.Logf("Sending a request to vtunnel peer process in [%v] over: 127.0.0.1%v", wslDistroName, peerTCPPort)
+	peerAddr := fmt.Sprintf("127.0.0.1:%v", peerTCPPort)
+	out, err := cmdRunWithOutput("wsl", "--distribution", wslDistroName, "--exec", "curl", peerAddr)
+	require.NoError(t, err, "Failed sending request to vtunnel peer process")
+	require.Contains(t, out, "vtunnel host called.")
+}
+
+func startHTTPServer(t *testing.T, ipPort string) {
```

Comment:
Consider using [`httptest.NewServer`](https://pkg.go.dev/net/http/httptest@go1.18.2#NewServer) too?

Also, we definitely shouldn't be supplying the port here…